extra/mariabackup/xtrabackup.cc

```diff
@@ -3589,18 +3590,9 @@ static bool xtrabackup_copy_logfile()
           msg("Error: write to ib_logfile0 failed");
           return true;
         }
-        else
-        {
-          const auto ofs= recv_sys.offset & ~block_size_1;
-          memmove_aligned<64>(log_sys.buf, log_sys.buf + ofs,
-                              recv_sys.len - ofs);
-          recv_sys.len-= ofs;
-          recv_sys.offset&= block_size_1;
-        }
-
         pthread_cond_broadcast(&scanned_lsn_cond);
 
-        if (r == recv_sys_t::GOT_EOF)
+        if (r == recv_sys_t::GOT_EOF || early_exit)
           break;
 
         if (recv_sys.offset < log_sys.write_size)
```

Comment:
I would suggest to add an assertion between the two `if` statements to document the case that we are handling:
```
ut_ad(r == recv_sys_t::PREMATURE_EOF);
```